addons/mail/models/mail_channel.py

```diff
@@ -1198,8 +1172,6 @@ def execute_command_help(self, **kwargs):
         partner = self.env.user.partner_id
         if self.channel_type == 'channel':
             msg = _("You are in channel <b>#%s</b>.", self.name)
-            if self.public == 'private':
-                msg += _(" This channel is private. People must be invited to join it.")
```

Comment:
I would think this condition should be moved for the channel type group case then?